code/game/objects/items/crab17.dm

```diff
@@ -0,0 +1,226 @@
+	if(!ishuman(user))
+		to_chat(user, "<span class='warning'>This device is too advanced for you!</span>")
+		return
+	if(dumped)
+		to_chat(user, "<span class='warning'>You already activated Protocol CRAB-17.</span>")
+		return FALSE
+	if(alert(user, "Are you sure you want to crash this market with no survivors?", "Protocol CRAB-17", "Yes", "No") == "Yes")
+		if(dumped || !src || QDELETED(src)) //Prevents fuckers from cheesing alert
+			return FALSE
+		var/turf/targetturf = get_random_station_turf()
+		new /obj/effect/dumpeetTarget(targetturf, user)
+		dumped = TRUE
+
+/obj/structure/checkoutmachine
+	name = "Nanotrasen Space-Coin Market"
+	desc = "This is good for spacecoin because"
+	icon = 'icons/obj/money_machine.dmi'
+	icon_state = "bogdanoff"
+	layer = TABLE_LAYER //So that the crate inside doesn't appear underneath
+	armor = list("melee" = 30, "bullet" = 50, "laser" = 50, "energy" = 100, "bomb" = 100, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 80)
+	density = TRUE
+	var/list/accounts_to_rob
+	var/mob/living/carbon/human/bogdanoff
+	var/canwalk = FALSE
+	pixel_z = -8
+	layer = LARGE_MOB_LAYER
+	max_integrity = 3000
+
+/obj/structure/checkoutmachine/examine(mob/living/user)
+	to_chat(user, "<span class='info'>It's integrated integrity meter reads: <b>HEALTH: [obj_integrity]</b>.</span>")
+
+/obj/structure/checkoutmachine/proc/check_if_finished()
+	for(var/i in accounts_to_rob)
+		var/datum/bank_account/B = i
+		if (B.being_dumped)
+			return FALSE
+	return TRUE
+
+/obj/structure/checkoutmachine/attackby(obj/item/W, mob/user, params)
+	if(check_if_finished())
+		qdel(src)
+		return
+	if(istype(W, /obj/item/card/id))
+		var/obj/item/card/id/card = W
+		if(!card.registered_account)
+			to_chat(user, "<span class='warning'>This card does not have a registered account!</span>")
+			return
+		if(!card.registered_account.being_dumped)
+			to_chat(user, "<span class='warning'>It appears that your funds are safe from draining!</span>")
+			return
+		if(do_after(user, 40, target = src))
+			if(!card.registered_account.being_dumped)
+				return
+			to_chat(user, "<span class='warning'>You quickly cash out your funds to a more secure banking location. Funds are safu.</span>")
+			card.registered_account.being_dumped = FALSE	
+			card.registered_account.canWithdraw = 0
+			if(check_if_finished())
+				qdel(src)
+				return
+	else
+		return ..()
+
+/obj/structure/checkoutmachine/Initialize(mapload, mob/living/user)
+	. = ..()
+	bogdanoff = user
+	add_overlay("flaps")
+	add_overlay("hatch")
+	add_overlay("legs_retracted")
+	addtimer(CALLBACK(src, .proc/startUp), 50)
+	START_PROCESSING(SSfastprocess, src)
+	
+/obj/structure/checkoutmachine/proc/startUp() //very VERY snowflake code that adds a neat animation when the pod lands.
+	start_dumping() //The machine doesnt move during this time, giving people close by a small window to grab their funds before it starts running around
+	sleep(10)
+	if(!src || QDELETED(src))

```

Comment:
https://github.com/tgstation/tgstation/blob/7ee0b49f8e80b67514689c15c75a166902ad1fc3/code/__DEFINES/qdel.dm#L22

you're checking for `!src` twice

Reply:
also lol this proc